autoprotocol/harness.py

```diff
@@ -200,6 +200,17 @@ def convert_param(protocol, val, type_desc):
                 f"The value supplied to input '{label}' (type aliquot++) is "
                 f"improperly formatted."
             )
+    elif type == "compound":
+        return str(val)
```

Comment:
We probably want some sort of first-class Compound class representation in the future. Ideally, we can actually add some validation on this (e.g. SMILES string).

Reply:
Found this: https://gist.github.com/lsauer/1312860/264ae813c2bd2c27a769d261c8c6b38da34e22fb

Reply:
Eh, I'm not confident in that regex. I'm ok with not validating for now.

Reply:
> We probably want some sort of first-class Compound class representation in the future

Agreed. Let's make a `Compound` class. 

The `val` should be a `dict` with a `notation` (possible values: `SMILES`, but maybe `InChI` in the future) and a `value`, similar to the strategy we adopted in here: https://github.com/strateos/web/blob/master/app/controllers/api/v1/inventory_searches_controller.rb#L85

Reply:
> We should actually add an ASC for Compound to the Autoprotocol Spec for the Derived Typesfield. See https://github.com/autoprotocol/ascs/blob/master/Specification/Definitions/types.md 

I'll make a small story for this. We should probably take this up internally, @yangchoo.